build: Install libraries in an `arch` sub-folder

Steelskin opened the pull request:
This is the proper installation scheme for Swift libraries and prevents having to manually copy them in `build.ps1`.

Steelskin commented:
@swift-ci please test

etcwilde commented on cmake/modules/SwiftSupport.cmake:
```diff
@@ -1,30 +1,67 @@
 include_guard()
 
-if(NOT dispatch_MODULE_TRIPLE)
+if(NOT dispatch_Swift_MODULE_TRIPLE OR NOT dispatch_Swift_ARCH OR NOT dispatch_Swift_SYSTEM)
```
I don't personally care that much as long as the project is internally consistent, though it does make the patch a little harder to read.

Steelskin replied:
The main reason is that the variable specifically applies to the Swift language. I have renamed the variables to lower the diff.

Steelskin commented:
@swift-ci please test windows platform

Steelskin commented:
@swift-ci please test

compnerd commented:
I think that we should do a cross-repo test and ensure that we can build the windows toolchain, Linux toolchain, and that they both function after this change.

Steelskin commented:
I tested the resulting installer on Windows and our internal projects built successfully with it. Can we land this or is there anything else I should test here?

Steelskin commented:
> Yes, please also validate that the Linux toolchain still works properly.

Thank you for the pointers for installing the built artifact. Here is a test project I used:

```
$ ~/usr/bin/swift test
[1/1] Planning build
Building for debugging...
/home/steelo/src/cassowary/Tests/CassowaryTests/CassowaryTests.swift:231:16: warning: 'add(constraint:_:)' is deprecated: renamed add(constraint:strength:) [#DeprecatedDeclaration]
229 |
230 |     try solver.add(variable: c, strength: .strong)
231 |     try solver.add(constraint: a >= 0, .strong)
    |                `- warning: 'add(constraint:_:)' is deprecated: renamed add(constraint:strength:) [#DeprecatedDeclaration]
232 |     try solver.add(constraint: b >= a, .strong)
233 |     try solver.add(constraint: b - a == c, .required)

/home/steelo/src/cassowary/Tests/CassowaryTests/CassowaryTests.swift:232:16: warning: 'add(constraint:_:)' is deprecated: renamed add(constraint:strength:) [#DeprecatedDeclaration]
230 |     try solver.add(variable: c, strength: .strong)
231 |     try solver.add(constraint: a >= 0, .strong)
232 |     try solver.add(constraint: b >= a, .strong)
    |                `- warning: 'add(constraint:_:)' is deprecated: renamed add(constraint:strength:) [#DeprecatedDeclaration]
233 |     try solver.add(constraint: b - a == c, .required)
234 |     try solver.suggest(value: 100, for: c)

/home/steelo/src/cassowary/Tests/CassowaryTests/CassowaryTests.swift:233:16: warning: 'add(constraint:_:)' is deprecated: renamed add(constraint:strength:) [#DeprecatedDeclaration]
231 |     try solver.add(constraint: a >= 0, .strong)
232 |     try solver.add(constraint: b >= a, .strong)
233 |     try solver.add(constraint: b - a == c, .required)
    |                `- warning: 'add(constraint:_:)' is deprecated: renamed add(constraint:strength:) [#DeprecatedDeclaration]
234 |     try solver.suggest(value: 100, for: c)
235 |

[#DeprecatedDeclaration]: <https://docs.swift.org/compiler/documentation/diagnostics/deprecated-declaration>
[23/23] Linking cassowaryPackageTests.xctest
Build complete! (4.67s)
Test Suite 'All tests' started at 2025-08-20 09:41:19.437
Test Suite 'debug.xctest' started at 2025-08-20 09:41:19.450
Test Suite 'CassowaryTests' started at 2025-08-20 09:41:19.450
Test Case 'CassowaryTests.testComplex1' started at 2025-08-20 09:41:19.450
Test Case 'CassowaryTests.testComplex1' passed (0.001 seconds)
Test Case 'CassowaryTests.testComplex2' started at 2025-08-20 09:41:19.451
Test Case 'CassowaryTests.testComplex2' passed (0.0 seconds)
Test Case 'CassowaryTests.testHandlingInfeasibleConstraints' started at 2025-08-20 09:41:19.451
Test Case 'CassowaryTests.testHandlingInfeasibleConstraints' passed (0.0 seconds)
Test Case 'CassowaryTests.testMultiTermExpressionOrder' started at 2025-08-20 09:41:19.452
Test Case 'CassowaryTests.testMultiTermExpressionOrder' passed (0.0 seconds)
Test Case 'CassowaryTests.testSimple1' started at 2025-08-20 09:41:19.452
Test Case 'CassowaryTests.testSimple1' passed (0.0 seconds)
Test Case 'CassowaryTests.testSimple2' started at 2025-08-20 09:41:19.452
Test Case 'CassowaryTests.testSimple2' passed (0.0 seconds)
Test Case 'CassowaryTests.testSimple3' started at 2025-08-20 09:41:19.452
Test Case 'CassowaryTests.testSimple3' passed (0.0 seconds)
Test Case 'CassowaryTests.testUnderConstrainedSystem' started at 2025-08-20 09:41:19.453
Test Case 'CassowaryTests.testUnderConstrainedSystem' passed (0.106 seconds)
Test Case 'CassowaryTests.testWithStrength' started at 2025-08-20 09:41:19.559
Test Case 'CassowaryTests.testWithStrength' passed (0.002 seconds)
Test Case 'CassowaryTests.testWithStrength2' started at 2025-08-20 09:41:19.561
Test Case 'CassowaryTests.testWithStrength2' passed (0.002 seconds)
Test Suite 'CassowaryTests' passed at 2025-08-20 09:41:19.562
         Executed 10 tests, with 0 failures (0 unexpected) in 0.113 (0.113) seconds
Test Suite 'debug.xctest' passed at 2025-08-20 09:41:19.562
         Executed 10 tests, with 0 failures (0 unexpected) in 0.113 (0.113) seconds
Test Suite 'All tests' passed at 2025-08-20 09:41:19.563
         Executed 10 tests, with 0 failures (0 unexpected) in 0.113 (0.113) seconds
◇ Test run started.
↳ Testing Library Version: 6.3-dev (a68a681c8adcd35)
↳ Target Platform: x86_64-unknown-linux-gnu
✔ Test run with 0 tests in 0 suites passed after 0.001 seconds.
```
